src/components/area-plot/AreaPlot.js

```diff
@@ -1,100 +1,183 @@
+            data={data}
+            dataChangeType={dataAnnotationsChangeType}
+            valueFormatter={yAxisTickFormatter}
+            getCurrentValue={(dataIndex, dataKey) => data[dataIndex][dataKey]}
+            getTotalValue={(dataIndex) =>
+              uniqueValuesOfCategoryKey.reduce(
+                (total, categoryKeyValue) => data[dataIndex][categoryKeyValue] + total,
+                0
+              )
+            }
+          />
+        ) : undefined
+      }
+    />
+  ));
+}
 
-  const yAxisDataKey = getYAxisDataKey(plotConfig);
-
+function NonPivotedAreaPlot({ plotConfig }) {
+  const categoryValueAxes = getCategoryValueAxes(plotConfig);
+  const showDataAnnotations = getSeriesShowDataAnnotations(plotConfig);
+  const dataAnnotationsChangeType = getAreaPlotDataAnnotationsChangeType(plotConfig);
+  const data = getData(plotConfig);
+  return categoryValueAxes.map((axis, index) => (
+    <Area
+      type="monotone"
+      dataKey={axis.dataKey}
+      name={axis.name}
+      key={axis.name}
+      fill={PLOT_SECONDARY_COLORS[index % PLOT_SECONDARY_COLORS.length]}
+      stroke={PLOT_COLORS[index % PLOT_COLORS.length]}
+      strokeWidth={2}
+      stackId="1"
+      isAnimationActive={false}
+      label={
+        showDataAnnotations ? (
+          <StackedDataAnnotation
+            data={data}
+            dataKey={axis.dataKey}
+            getCurrentValue={(dataIndex, dataKey) => data[dataIndex][dataKey]}
+            getTotalValue={(dataIndex) =>
+              categoryValueAxes.reduce(
+                (total, categoryValueAxis) => data[dataIndex][categoryValueAxis.dataKey] + total,
+                0
+              )
+            }
+            dataChangeType={dataAnnotationsChangeType}
+            valueFormatter={getTickFormatterFromDataKey(plotConfig, axis.dataKey)}
+          />
+        ) : undefined
+      }
+    />
+  ));
+}
+function AreaPlot({
+  plotConfig = {},
+  TooltipContent = () => {},
+  onBrushUpdate = () => {},
+  isFollowUpDisabled = false,
+}) {
   const data = getData(plotConfig);
   const margin = getMargin(plotConfig);
+  const xAxisDataKey = getXAxisDataKey(plotConfig);
+  const xAxisFormat = getAxisFormat(plotConfig, xAxisDataKey);
 
-  const categoryValueAxes = getCategoryValueAxes(plotConfig);
-  const isDataPivoted = getIsDataPivoted(plotConfig);
-  return (
-    <ResponsiveContainer>
-      <AreaChart margin={margin} data={data}>
-        {GridLines()}
+  const [tooltip, tooltipHandlers] = useTooltip();
+  const [brush, brushEvents] = useBrush({
+    onBrushUpdate,
+    tooltipHandlers,
+    tooltip,
+    xAxisDataKey,
+    xAxisFormat,
+    brushSelectionType: BRUSH_SELECTION_TYPES.RANGE_AND_ITEMS,
+  });
 
-        {XAxis({ ...xAxisConfig })}
-        {YAxis({})}
+  const plotDataChangeType = getAreaPlotDataChangeType(plotConfig);
+  const isDataPivoted = getIsDataPivoted(plotConfig);
+  const fullYAxisConfig = getYAxis(plotConfig);
```

Comment:
just change to yAxisConfig, like funnelbarplot

Reply:
Can we break this into a reusable function and update all of the other plots to use it.
overrideAxisConfig(axis, overides)